libraries/AP_Rally/AP_Rally.cpp

```diff
@@ -161,3 +172,40 @@ Location AP_Rally::calc_best_rally_or_home_location(const Location &current_loc,
 
     return return_loc;
 }
+// return best rally location from current position - not including home or km limit
+Location AP_Rally::calc_best_rally_location(const Location &current_loc, float rtl_home_alt) const
```

Comment:
same calc_best_rally_or_home_location ?

Reply:
nope, this one exclude home from calc but it still verify its validity (rally point inside fence) otherwise it replace by home.

Reply:
`find_nearest_rally_point` returns false when home is nearest than any rally points or when all rally points are further than the `_rally_limit_km`

What do you mean by changing the methods instead of using the parameter?
Do you want me to change the  `find_nearest_rally_point` method to have an extra argument for including home or not ?